frontend/lib/item_listing/view/listing.view.dart

```diff
@@ -0,0 +1,71 @@
+import 'package:flutter/material.dart';
+import 'package:provider/provider.dart';
+import 'package:utm_marketplace/item_listing/components/item_card/item_card.component.dart';
+import 'package:utm_marketplace/item_listing/view_models/listing.viewmodel.dart';
+import 'package:utm_marketplace/shared/components/loading.component.dart';
+
+class ListingView extends StatefulWidget {
+  const ListingView({super.key});
+
+  @override
+  State<ListingView> createState() => _ListingViewState();
+}
+
+class _ListingViewState extends State<ListingView> {
+  late ListingViewModel viewModel;
+
+  @override
+  void initState() {
+    super.initState();
+    viewModel = Provider.of<ListingViewModel>(context, listen: false);
+
+    WidgetsBinding.instance.addPostFrameCallback((_) {
+      viewModel.fetchData();
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(
+        title: const Text('Items Listing'),
+      ),
+      body: SafeArea(
+        child: Consumer<ListingViewModel>(
+          builder: (_, model, child) {
+            if (model.isLoading) {
+              return const Center(child: LoadingComponent());
+            }
+
+            if (model.listingModel.items.isEmpty) {
```

Comment:
You shouldn't directly access the models from the ViewModel. In MVVM, the ViewModel mediates all interactions between View and Model. I'll discuss this more in the meeting and on Discord.